### PR TITLE
AIP-59-performance-dags

### PR DESCRIPTION
**Pull request by bjankie1**
Part of AIP-59. The change is focused on introducing a simple framework that constructs DAGs of different size and shape that will be used in evaluation of performance.

---
**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/contributing-docs/05_pull_requests.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

**Comment by uranusjr** [comment]
Lots of linter errors need to be fixed, but code-wise looks good to me.

File: performance/src/performance_dags/elastic_dag/elastic_dag.py [hunk 1]
```diff
@@ -0,0 +1,322 @@
+"""
+Elastic dag copied from apache/airflow master and adjusted to work on python2 and with Airflow
```
Comment (potiuk) [inline performance/src/performance_dags/elastic_dag/elastic_dag.py]:
```suggestion
Elastic dag copied from apache/airflow main and adjusted to work on python2 and with Airflow
```

Comment (potiuk) [reply performance/src/performance_dags/elastic_dag/elastic_dag.py]: Also -  I think that comment is outdated in general.

File: performance/src/performance_dags/elastic_dag/elastic_dag.py [hunk 2]
```diff
@@ -0,0 +1,322 @@
...
+                **operator_extra_kwargs,
+            )
+            for i in range(1, task_count + 1)
+        ]
+    else:
+        raise ValueError(f"Unsupported operator type: {operator_type_str}.")
+    return task_list
+
+
+def chain(*tasks):
```
Comment (potiuk) [inline performance/src/performance_dags/elastic_dag/elastic_dag.py]: I think that was used for Airflow 1->2  migration when we did not have `chain` but I think chain is now defined in BaseOperator, we do not need to copy it here I **think**.

**Comment by potiuk** [comment]
I thnk it also needs a bit cleanup for outdated stuff - generally modernising the code and see what can be reused from plain Airflow 2 perspective. 

Also possibly we should renamae the "elastic" dag into something else (`flexible` maybe`) - for me `elastic` is a bit confusing with `elasticsearch` and I was wondering for a while why we copied things from there.

File: performance/src/performance_dags/elastic_dag/elastic_dag.py (hunk 1 above)
Comment (bjankie1) [reply performance/src/performance_dags/elastic_dag/elastic_dag.py]: Agree. I have updated the comment adding more information about the DAG usage.

File: performance/src/performance_dags/elastic_dag/elastic_dag.py (hunk 2 above)
Comment (bjankie1) [reply performance/src/performance_dags/elastic_dag/elastic_dag.py]: Indeed. I used `BaseOperator.chain` reference instead.

**Comment by bjankie1** [comment]
> I thnk it also needs a bit cleanup for outdated stuff - generally modernising the code and see what can be reused from plain Airflow 2 perspective.
> 
> Also possibly we should renamae the "elastic" dag into something else (`flexible` maybe?) - for me `elastic` is a bit confusing with `elasticsearch` and I was wondering for a while why we copied things from there.

Renamed to `performance_dag`.  I will look for some other opportunities to modernize the code. I'd also welcome your suggestions for that.

**Comment by bjankie1** [comment]
@potiuk checks are passing :)

**Comment by uranusjr** [comment]
One check is not.

**Comment by potiuk** [comment]
Yeah. We need more static check fixes for that one.

**Comment by bjankie1** [comment]
> Yeah. We need more static check fixes for that one.

Fixed all of them I believe except for `assert` in the test complains. I believe it requires a check modification to exclude it in this folder.

**Comment by potiuk** [comment]
Yeah you can exclude them in this PR. This is easiest and makes most sense

**Comment by bjankie1** [comment]
> One check is not.

fixed

**Review by potiuk — APPROVED** [review]

**Pull request merged**